datafusion/common/src/scalar.rs

```diff
@@ -221,17 +221,9 @@ impl PartialOrd for ScalarValue {
             (Decimal128(_, _, _), _) => None,
             (Boolean(v1), Boolean(v2)) => v1.partial_cmp(v2),
             (Boolean(_), _) => None,
-            (Float32(v1), Float32(v2)) => {
-                let v1 = v1.map(OrderedFloat);
-                let v2 = v2.map(OrderedFloat);
-                v1.partial_cmp(&v2)
-            }
+            (Float32(v1), Float32(v2)) => v1.partial_cmp(v2),
```

Comment:
```suggestion
            (Float32(v1), Float32(v2)) => v1.total_cmp(v2),
```

Reply:
v1 is an `Option<f32>`, it supports partial_cmp, not total_cmp. let me know if you ok if I unwrap it to floats, the same way as done for Decimals.

Reply:
Yes, we will need to unwrap the option, I keep forgetting that ScalarValue has typed nulls for some reason :laughing: 

partial_cmp on Option, will call through to partial_cmp on f32, which is not the same as total_cmp

Reply:
Done. Yeah, I checked that Float64(NULL) == Float64(NULL) now.